code/game/objects/item.dm

```diff
@@ -684,17 +684,24 @@ var/global/list/slot_flags_enumeration = list(
 
 /obj/item/add_blood(mob/living/carbon/human/M, amount = 2, blood_data)
 	if (!..())
-		return 0
+		return FALSE
 
 	if(istype(src, /obj/item/energy_blade))
 		return
 
-	if(!blood_data && istype(M))
-		blood_data = M.vessel.reagent_data[/decl/material/liquid/blood]		
-	var/datum/extension/forensic_evidence/forensics = get_or_create_extension(src, /datum/extension/forensic_evidence)
-	forensics.add_data(/datum/forensics/blood_dna, blood_data["blood_DNA"])
-	add_coating(/decl/material/liquid/blood, amount, blood_data)
-	return 1 //we applied blood to the item
+	if(blood_data || !istype(M))
```

Comment:
Won't this prevent extra blood DNA being added to `blood_data`?

Reply:
No, tested in game

Reply:
I got this around the wrong way - this code means that if a non-null blood_data is supplied to the proc (which is done in at least one spot), nothing will ever be added to it. This isn't the same as the previous code where blood_data not being provided to the proc results in it being set to the reagent data.

Also I suspect that the lazyset on blood_data is mutating the reagent data list.